core/conn/jdbcT4/src/main/java/org/trafodion/jdbc/t4/Bytes.java

```diff
@@ -101,19 +103,16 @@ static long extractLong(byte[] array, int offset, boolean swap) {
 		long value = 0;
 		int i=offset;
 		
-		if(swap) {
-			for (int shift = 0; shift < 64; shift += 8) {
-				value |= ( (long)( array[i] & 0xff ) ) << shift;
-				i++;
-			}
-			
-		}else {
-			for (int shift = 56; shift >= 0; shift -= 8) {
-				value |= ( (long)( array[i] & 0xff ) ) << shift;
-				i++;
-			}
-		}
-		
+        ByteBuffer buffer = ByteBuffer.allocate(8);
```

Comment:
Can you please consider using ByteBuffer.wrap(array, offset, 8) to avoid allocation and then copy the data. In addition, it is not clear what was the problem with the original code. A test case or an explanation how it failed would be good to understand the issue. My understanding is that shift operator would work fine with long data type too.